timings_of_new_method

```diff
@@ -0,0 +1,32 @@
+│  64 │ 0.3 │   0.5 │    1.559 s │    1.779 s │    1.722 s │    1.828 s │   1.73 GiB │   8969362 │       3 │
+│  64 │ 0.6 │   0.5 │ 411.863 ms │ 430.678 ms │ 425.296 ms │ 433.009 ms │ 478.89 MiB │   1296680 │       5 │
+│  64 │ 0.1 │   2.0 │    4.711 s │    4.748 s │    4.748 s │    4.786 s │   5.20 GiB │  40335010 │       2 │
+│  64 │ 0.3 │   2.0 │    1.326 s │    1.341 s │    1.344 s │    1.368 s │   1.73 GiB │   8969362 │       4 │
+│  64 │ 0.6 │   2.0 │ 388.265 ms │ 414.406 ms │ 413.665 ms │ 449.928 ms │ 478.89 MiB │   1296680 │       5 │
+│  64 │ 0.1 │   1.0 │    1.328 s │    1.354 s │    1.356 s │    1.387 s │   1.73 GiB │   8969362 │       4 │
+│  64 │ 0.3 │   1.0 │    1.283 s │    1.290 s │    1.310 s │    1.376 s │   1.73 GiB │   8969362 │       4 │
+│  64 │ 0.6 │   1.0 │ 387.341 ms │ 389.817 ms │ 395.236 ms │ 411.333 ms │ 478.89 MiB │   1296680 │       5 │
+│ 128 │ 0.1 │   0.5 │   13.551 s │   13.551 s │   13.551 s │   13.551 s │  18.93 GiB │  35905046 │       1 │
+│ 128 │ 0.3 │   0.5 │   10.783 s │   10.783 s │   10.783 s │   10.783 s │  18.93 GiB │  35905046 │       1 │
+│ 128 │ 0.6 │   0.5 │    4.235 s │    4.254 s │    4.254 s │    4.273 s │   6.39 GiB │   5205036 │       2 │
+│ 128 │ 0.1 │   1.0 │   10.172 s │   10.172 s │   10.172 s │   10.172 s │  18.93 GiB │  35905046 │       1 │
+│ 128 │ 0.3 │   1.0 │    9.628 s │    9.628 s │    9.628 s │    9.628 s │  18.93 GiB │  35905046 │       1 │
+│ 128 │ 0.6 │   1.0 │    3.839 s │    3.871 s │    3.871 s │    3.904 s │   6.39 GiB │   5205036 │       2 │
+│ 128 │ 0.1 │   2.0 │   26.066 s │   26.066 s │   26.066 s │   26.066 s │  46.34 GiB │ 161396774 │       1 │
+│ 128 │ 0.3 │   2.0 │    9.690 s │    9.690 s │    9.690 s │    9.690 s │  18.93 GiB │  35905046 │       1 │
+│ 128 │ 0.6 │   2.0 │    3.768 s │    3.775 s │    3.775 s │    3.783 s │   6.39 GiB │   5205036 │       2 │
+│ 256 │ 0.1 │   0.5 │  139.427 s │  139.427 s │  139.427 s │  139.427 s │ 268.07 GiB │ 143648538 │       1 │
+│ 256 │ 0.3 │   0.5 │  113.194 s │  113.194 s │  113.194 s │  113.194 s │ 268.07 GiB │ 143648538 │       1 │
+│ 256 │ 0.6 │   0.5 │   52.846 s │   52.846 s │   52.846 s │   52.846 s │  97.95 GiB │  20838448 │       1 │
+│ 256 │ 0.1 │   1.0 │  109.350 s │  109.350 s │  109.350 s │  109.350 s │ 268.07 GiB │ 143648538 │       1 │
+│ 256 │ 0.3 │   1.0 │  106.777 s │  106.777 s │  106.777 s │  106.777 s │ 268.07 GiB │ 143648538 │       1 │
+│ 256 │ 0.6 │   1.0 │   52.511 s │   52.511 s │   52.511 s │   52.511 s │  97.95 GiB │  20838448 │       1 │
+│ 256 │ 0.1 │   2.0 │  246.468 s │  246.468 s │  246.468 s │  246.468 s │ 593.73 GiB │ 645644595 │       1 │
+│ 256 │ 0.3 │   2.0 │  104.747 s │  104.747 s │  104.747 s │  104.747 s │ 268.07 GiB │ 143648538 │       1 │
+│ 256 │ 0.6 │   2.0 │   49.531 s │   49.531 s │   49.531 s │   49.531 s │  97.95 GiB │  20838448 │       1 │
+└─────┴─────┴───────┴────────────┴────────────┴────────────┴────────────┴────────────┴───────────┴─────────┘
```

Comment:
Do we want this text file as part of the PR?

Reply:
ah no that was not supposed to be there! deleted it